libs/execution/src/lib/transforms/transform-executor.spec.ts

```diff
@@ -0,0 +1,371 @@
+// SPDX-FileCopyrightText: 2023 Friedrich-Alexander-Universitat Erlangen-Nurnberg
+//
+// SPDX-License-Identifier: AGPL-3.0-only
+
+import assert = require('assert');
+
+import {
+  InternalValueRepresentation,
+  PrimitiveValuetypes,
+  TransformDefinition,
+  createJayveeServices,
+  useExtension,
+} from '@jvalue/jayvee-language-server';
+import {
+  ParseHelperOptions,
+  TestLangExtension,
+  expectNoParserAndLexerErrors,
+  parseHelper,
+  readJvTestAssetHelper,
+} from '@jvalue/jayvee-language-server/test';
+import { AstNode, AstNodeLocator, LangiumDocument } from 'langium';
+import { NodeFileSystem } from 'langium/node';
+
+import { constructTable, getTestExecutionContext } from '../../../test/utils';
+import { Table, TableColumn } from '../types/io-types/table';
+
+import { PortDetails, TransformExecutor } from './transform-executor';
+
+describe('Validation of TransformExecutor', () => {
+  let parse: (
+    input: string,
+    options?: ParseHelperOptions,
+  ) => Promise<LangiumDocument<AstNode>>;
+
+  let locator: AstNodeLocator;
+
+  const readJvTestAsset = readJvTestAssetHelper(
+    __dirname,
+    '../../../test/assets/',
+  );
+
+  function getColumnsMap(
+    inputColumnNames: string[],
+    inputTable: Table,
+    transformInputDetailsList: PortDetails[],
+  ): Map<string, TableColumn> {
+    const variableToColumnMap: Map<string, TableColumn> = new Map();
+    for (let i = 0; i < inputColumnNames.length; ++i) {
+      const inputColumnName = inputColumnNames[i];
+      assert(inputColumnName !== undefined);
+      const inputColumn = inputTable.getColumn(inputColumnName);
+      assert(inputColumn !== undefined);
+
+      const matchingInputDetails = transformInputDetailsList[i];
+      assert(matchingInputDetails !== undefined);
+
+      const variableName = matchingInputDetails.port.name;
+      variableToColumnMap.set(variableName, inputColumn);
+    }
+    return variableToColumnMap;
+  }
+
+  async function parseAndExecuteTransform(
+    input: string,
+    inputTable: Table,
+    columnNames: string[],
+  ): Promise<{
+    resultingColumn: TableColumn<InternalValueRepresentation>;
+    rowsToDelete: number[];
+  }> {
+    const document = await parse(input, { validationChecks: 'all' });
+    expectNoParserAndLexerErrors(document);
+
+    const transform = locator.getAstNode<TransformDefinition>(
+      document.parseResult.value,
+      'transforms@0',
```

Comment:
Could you pls iterate over all transforms? Makes the test setup less brittle.

Reply:
That would most likely not change anything because the execution of the other transforms will most likely crash due to wrong input / execution context

Reply:
Discussed in person